### DISCUSSION TITLE
When should we turn on testing?

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#62] [Q&A] [answered]
Currently in our automated build I have purposely not included a task to run our unit tests. That's because so many are failing , we know that, and I think our build system reminding us every commit would be rubbing salt in the wound.

That said, I'm thinking about when it's useful to turn that back on. Once we have LP3 tests working, do we want to turn it on to verify we haven't screwed that up as we keep moving forward? 

Some critical factors:
- I can likely target the tests to specific projects (even though our whole solution doesn't pass, if we wanted to just test specific projects, we could do that.

-  All the tests currently in the solution are not necessary for achievement of the MVP, so we might not want to spend our time on fixing them.  

Some alternatives I've been considering:
1- Don't turn it on until it's all done. This is the easiest path forward, we can just trust eachother to run on our own machines before we commit to make sure we've not broken one of the tests that should be working. 

2- Turn it on and accept the red X's. Github will tell us we're failures every time, but we could read the actions log to see what tests  failed, and manually verify that a test we are worried about did pass on the server. 

3- Comment out/Unload test classes we don't need yet, and turn testing on. This feels bad, as I've had a generally negative feeling about seeing classes entirely commented out, or unloaded/compile excluded, but it is a way we could keep known failing tests out of our automated tests, and get the value of having the supposed-to-be-working tests running on the build system.

**Comment by HenryGeorgist** [comment]
I dont like alternative 1

I would favor alternative 3. I would remove the code and rely on version control to add them back in as they get added back in as we get them passing. We should be cautious about adding code until we have corrected tests in that area.

thoughts?

**Comment by rnugent3** [comment]
I'd lean toward alternative 3, too. I'd like to see automated tests running
on the classes we're working on asap.  Would we only remove the test
classes that we don't need and that are failing miserably, while leaving in
test classes we don't yet need but are passing?

**Comment by HenryGeorgist** [comment]
if tests are truly passing a test we should not remove it.

**Comment by Brennan1994** [comment]
Alright. I'm good with Alternative 3, and happy to kill more files in this repo. So path forward will be to delete any test classes which do not have passing tests, and any individual tests which do not yet pass. Once that's done and all existing tests pass on our machines, I'll turn on testing on our build system.

**Comment by HenryGeorgist** [comment]
make the removal of tests very easy to track in the issue tracking and commit names

so create an issue "Removing Not Passing tests for XXX" put a label (maybe we even make a special label) and then reference the issue with the commit and add text like "Removing Not Passing Tests for XXX addresses XXXX"

We definitely want to be able to query them easily later. So a common key word or something like a label (or both) will be super helpful.

**Comment by rnugent3** [comment]
Thanks, Brennan. Is it worth writing down somewhere the commit on which the
files are deleted or the list of the files deleted? Or can we just do a
search when we're ready to go back?

**Comment by rnugent3** [comment]
Woops! Will, looks like you answered my question.

**Comment by Brennan1994** [comment, marked as answer]
Alright, issues are created for each project. I created a label to associate with the issue called "remove old tests." I did not assign them all to me, so If you're hungry for things to do today feel free to pick them from the waiver wire. They're in the to-do column of our Kanban

Remember to close the issues with your commits. Removes old test would be a good first three words to include in the commit message for easier browsing later. 

Once that's done, I'll add unit testing, and we can add another stripe to our cool kidz rankings.
